src/app/frontend/deploy/deployfromsettings_directive.js

```diff
@@ -0,0 +1,33 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import DeployFromSettingsController from './deployfromsettings_controller';
+
+/**
+ * Returns directive definition object for the deploy from settings directive.
+ * @return {!angular.Directive}
+ */
+export default function deployFromSettingsDirective() {
+  return {
+    scope: {},
```

Comment:
Case1: If I don't use scope at all: It means `scope: false`
* Then I can't use `controllerAs: 'ctrl'`  in child directive. Because ctrl is already taken by parent controlller. If I use it anyway, then the ctrl from child directive overwrites the one from parent. The child ctrl should be %100 extending the parent ctrl to make it work even if it overwrites.  
* If I use different name like 'ctrlSettings' then directive-parameter binding doesn't work.

Case2: If I use `scope: true`

* Then still I can't use `controllerAs: 'ctrl'` because it creates a collision with parent ctrl and two ways directive-parameter binding doesn't work. I need to name it different again.

Case3: If I use `scope:{}` 

* Then I am free to use `controllerAs: 'ctrl'` and scope binding works fine. Nothing collides.